Add FreeBSD service (#2333)
sopel-irc/sopel: contrib/FreeBSD/README.md

This folder contains `sopel` and `sopel-default.cfg` designed to be distributed by third parties such as FreeBSD.

`sopel-default.cfg` is a default configuration file for sopel.

The -default prefix is mandatory and will be used as a 'profile'.

The profiles may be changed with the variable `sopel_profiles` in `/etc/rc.conf`. For example:

```sh
sysrc sopel_profiles="profile1 profile2 profile3"
```

By default, the configuration directory is `/usr/local/etc`.

> The `sopel-profile1.cfg`, `sopel-profile2.cfg` and `sopel-profile3.cfg` files must be stored in the configuration directory.

Reviewer: ```suggestion
For the above example, the configuration files `sopel-profile1.cfg`, `sopel-profile2.cfg`, and `sopel-profile3.cfg` must be present there.
```
Conditional on the example - otherwise I can see someone reading it as mandatory.

> sopel- is mandatory and can be changed with the `sopel_prefix` variable in `/etc/rc.conf` as another FreeBSD service.

Author: I will remove the `sopel_prefix`, it is intended for packagers, but I suggest that `sopel_interpreter` should be kept because, on FreeBSD, there is no `/usr/local/bin/python` link to `/usr/local/bin/python3.8|3.9|x.xx`, so it is necessary to reconfigure `sopel_interpreter` when python has a new version on FreeBSD. The `sopel_interpreter` variable is not only used to run sopel, it is also used to check if sopel is running with the `check_pidfile` function.

The service must be installed in `/usr/local/etc/rc.d`:

```sh
cp sopel /usr/local/etc/rc.d
chmod +x /usr/local/etc/rc.d/sopel
```

The default configuration file must be installed in the configuration directory as another profile. For example:

```sh
cp sopel-default.cfg /usr/local/etc
```

If you want to run sopel at startup, the variable `sopel_enable` must be set to `YES`:

```sh
sysrc sopel_enable="YES"
```

> If you want to change the python version, the variable `sopel_interpreter` must be changed. For example:

Reviewer: Necessary for end user docs? How common is this need?

Reviewer: https://github.com/sopel-irc/sopel/pull/2333#discussion_r930317048
